clang/lib/Sema/SemaBounds.cpp

```diff
@@ -4017,11 +4023,55 @@ namespace {
         << ObservedBounds << ObservedBounds->getSourceRange();
     }
 
+    // BlameAssignmentWithinStmt prints a diagnostic message that highlights the
+    // assignment expression in St that causes V's observed bounds to be unknown
+    // or invalid.  If St is a DeclStmt, St itself and V are highlighted.
```

Comment:
Nit: BlameAssignmentWithinStmt can also be used to print a diagnostic message if V's bounds are not necessarily invalid, but are not provably valid. I suggest updating the comment to something like "not provably valid" instead of "invalid".

Reply:
Thanks for the suggestion!